parsl/executors/flux/execute_parsl_task.py

```diff
@@ -0,0 +1,43 @@
+"""Script for executing tasks inside of Flux jobs."""
+
+import argparse
+import os
+import logging
+
+from parsl.executors.high_throughput.process_worker_pool import execute_task
+from parsl.serialize import serialize
+
+
+def main():
+    """Execute one rank of an MPI application."""
+    logging.basicConfig(
+        level=logging.INFO, format="%(asctime)s [%(levelname)s] %(message)s"
+    )
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-i", "--input", required=True, help="Input pickle file")
+    parser.add_argument("-o", "--output", required=True, help="Output pickle file")
+    args = parser.parse_args()
+    logging.info("Input : %s", args.input)
+    logging.info("Output : %s", args.output)
+    result = None
+    # open and deserialize the task's pickled input package
+    with open(args.input, "rb") as f:
+        fn_buf = f.read()
+        logging.info("Read input pickle file")
+        try:
+            result = execute_task(fn_buf)
+        except Exception as e:
+            logging.exception("Parsl task execution failed:")
+            result = e
```

Comment:
As it stands, there is no way to distinguish between an exception thrown by a callable and one returned. So if the callable raises an exception, you would fetch it with ``future.result()`` and not ``future.exception()`` as one might expect. Is this behavior consistent with the other Parsl executors? I'm guessing not, just want to make sure.

Reply:
If the call raised, `future.result()` will raise the same exception. This is consistent with the `concurrent.futures` behavior: https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Future.result. `future.exception()` on the other hand returns the exception and does not raise it.

Reply:
Right, that's what I thought. I think I inherited this code from somewhere else and just wanted to make sure it worked correctly. Thanks for clarifying!

Reply:
I think this behavior is not consistent with the general contract of `Future`. The idea being that `Future.result()` should behave as if the computation wrapped by the future was invoked synchronously. That is, if said computation  throws an exception instead of returning a value, so should `Future.result()`.

Reply:
Right, I think we're all agreed on what's supposed to happen.

Reply:
Added a commit that should resolve this problem.